app/models/user.rb

```diff
@@ -900,10 +899,17 @@ def like_given_count
   end
 
   def post_count
-    stat = user_stat || create_user_stat
     stat.post_count
   end
 
+  def post_edits_count
+    stat.post_edits_count
+  end
+
+  def increment_post_edits_count
+    stat.update!(post_edits_count: stat.post_edits_count + 1)
```

Comment:
The fetching of `post_edits_count` and updating of the count needs to happen in a transaction. Otherwise, a race condition can happen where another process may have updated the count after we've fetch it from the database.

Reply:
It looks like this method is currently only called from inside a transaction from `PostRevisor`. 

```suggestion
    stat.increment!(:post_edits_count)
```

Still I think this query is better as it will not be subjected to concurrency issues if someone accidentally calls it outside a transaction in the future.